pkg/gardenlet/controller/shoot/shoot_control_reconcile.go

```diff
@@ -446,28 +455,31 @@ func (c *Controller) runReconcileShootFlow(o *operation.Operation) *gardencorev1
 		return gardencorev1beta1helper.NewWrappedLastErrors(gardencorev1beta1helper.FormatLastErrDescription(err), flow.Errors(err))
 	}
 
-	// Remove completed tasks from Shoot if they were executed.
+	// ensure that shoot client is invalidated after it has been hibernated
+	if o.Shoot.HibernationEnabled {
+		if err := o.ClientMap.InvalidateClient(keys.ForShoot(o.Shoot.Info)); err != nil {
+			err = fmt.Errorf("failed to invalidate shoot client: %w", err)
+			return gardencorev1beta1helper.NewWrappedLastErrors(gardencorev1beta1helper.FormatLastErrDescription(err), err)
+		}
+	}
+
+	o.Logger.Infof("Successfully reconciled Shoot %q", o.Shoot.Info.Name)
+	return nil
+}
+
+func (c *Controller) removeTaskAnnotation(o *operation.Operation, generation int64, tasksToRemove ...string) error {
```

Comment:
This func currently is not using anything from the receiver.

```suggestion
func removeTaskAnnotation(o *operation.Operation, generation int64, tasksToRemove ...string) error {
```